src/diagnostic/diagnostic_writer.h

```diff
@@ -15,15 +15,6 @@ class TypeWriter
     virtual ~TypeWriter() {}
```

Comment:
is this not redundant also?

Reply:
you mean the abstract class as a whole?
it's actually used...